src/System/refine.rs

```diff
@@ -854,46 +438,31 @@ fn prepare_init_state(
 }
 
 fn precheck_sys_rep(sys: &mut SystemRepresentation) -> bool {
```

Comment:
we could implement this on SystemRepresentation instead as it seems like a general thing that we'll need to check independent of the type of query